cuda_core/cuda/core/experimental/_program.py

```
@@ -440,6 +458,12 @@ def compile(self, target_type, name_expressions=(), logs=None):
             raise NotImplementedError
 
         if self._backend == "nvrtc":
+            version = handle_return(nvrtc.nvrtcVersion())
+            if handle_return(driver.cuDriverGetVersion()) > version[0] * 1000 + version[1] * 10:
+                raise RuntimeError(
+                    "The CUDA driver version is newer than the NVRTC version. "
+                    "Please update your NVRTC library to match the CUDA driver version."
+                )
```

Unfortunately this makes our run-time requirements unnecessarily strict. Driver is always backward compatible so new driver + old CTK (on a device that the old CTK supports) should work just fine.

In the test file there's a removed comment. If this part is meant to follow it, it should check whether target_type is PTX and then compare the driver and NVRTC versions. PTX generated by new NVRTC/NVVM/nvJitLink cannot be loaded (JIT-compiled) by old driver.

I think we should just move around that can_load_generated_ptx check to the codebase, and reuse it here. Then, in the test suite if the local/CI environment does not meet the requirement, we handle the raised exception by the check, instead of skipping the test. Does it make sense?

---

I switched to a warning since the user is fine to transpile their PTX, but should be aware that they will run into problems when they try to load it